lib/runtime/storage/storagediff.go

```diff
@@ -141,13 +141,17 @@ func (cs *storageDiff) clearPrefixInChild(keyToChild string, prefix []byte,
 // optional limit. It returns the number of keys deleted and a boolean
 // indicating if all keys with the prefix were removed.
 func (cs *storageDiff) clearPrefix(prefix []byte, trieKeys []string, limit int) (deleted uint32, allDeleted bool) {
-	allKeys := slices.Clone(trieKeys)
 	newKeys := maps.Keys(cs.upserts)
-	allKeys = append(allKeys, newKeys...)
+	keysToClear := newKeys
```

Comment:
you can use the slice that `maps.Keys` return, no need to declare a new variable
```suggestion
	keysToClear := maps.Keys(cs.upserts)
```